### PR TITLE
Refactor k8s IT checks to make logic more obvious

### PR DESCRIPTION
**Pull request by nuwang**
This was missed in the previous refactoring run here: https://github.com/galaxyproject/galaxy/pull/12285

## How to test the changes?
(Select all options that apply)
- [ ] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [X] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these contributions under [Galaxy's current license](https://github.com/galaxyproject/galaxy/blob/dev/LICENSE.txt).
- [x] I agree to allow the Galaxy committers to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT). If this condition is an issue, uncheck and just let us know why with an e-mail to galaxy-committers@lists.galaxyproject.org.

### CODE REVIEW

**Review by natefoo — APPROVED** [review]

**Comment by jmchilton** [comment]
This seems backwards to me - I think the thing that should inspire the cleanup is having guest ports in the tool's container to cleanup. Right now, those only happen in interactive tools but I've setup most of the job and tool logic outside the Kubernetes runner around those guest_ports because that is the key difference and we might someday have other tool types that expose guest ports.

**Comment by mvdbeek** [comment]
@nuwang any chance you can adjust the naming or should we leave this untouched ?

**Comment by mvdbeek** [comment]
@nuwang I'm gonna bump it for now. Since that's targeted at developers I don't think we necessarily need to get this into 21.09

**Review by mvdbeek — APPROVED** [review]

**Comment by nuwang** [comment]
@mvdbeek That makes sense and sorry I couldn't attend to this earlier. I don't feel the illusion of using this terminology is that strong at the moment - there are still multiple places in the code that refer to interactive tools, such as `configure_interactive_tool_entry_points`. We'll probably need to change those as well in the long term.

**Comment by mvdbeek** [comment]
That looks related:
```
Traceback (most recent call last):
  File "/home/runner/work/galaxy/galaxy/galaxy root/lib/galaxy/jobs/runners/kubernetes.py", line 834, in stop_job
    if self.__has_guest_ports(job_wrapper.get_state()):
  File "/home/runner/work/galaxy/galaxy/galaxy root/lib/galaxy/jobs/runners/kubernetes.py", line 195, in __has_guest_ports
    return bool(ajs.job_wrapper.guest_ports)
AttributeError: 'str' object has no attribute 'job_wrapper'
```

File: lib/galaxy/jobs/runners/kubernetes.py [hunk 1]
```diff
@@ -826,8 +826,8 @@ def stop_job(self, job_wrapper):
             job_to_delete = find_job_object_by_name(self._pykube_api, job.get_job_runner_external_id(), self.runner_params['k8s_namespace'])
             if job_to_delete and len(job_to_delete.response['items']) > 0:
                 k8s_job = Job(self._pykube_api, job_to_delete.response['items'][0])
-                if job_wrapper.guest_ports:
-                    self.__cleanup_k8s_interactivetools(job_wrapper, k8s_job)
+                if self.__has_guest_ports(job_wrapper.get_state()):
```
Comment (mvdbeek) [inline lib/galaxy/jobs/runners/kubernetes.py]: `job_wrapper.get_state())` returns the state column of the job. Maybe refactor `__has_guest_ports` to take the job_wrapper directly ?

**Comment by mvdbeek** [comment]
@nuwang I've put this back into WIP and changed the milestone, but if you can fix this we can still get it into 21.09.

File: lib/galaxy/jobs/runners/kubernetes.py (hunk 1 above)
Comment (nuwang) [reply lib/galaxy/jobs/runners/kubernetes.py]: Thanks for pointing that out @mvdbeek. Have fixed it now.

**Pull request merged**